components/blitz/src/omero/cmd/admin/CurrentSessionsRequestI.java

```diff
@@ -0,0 +1,133 @@
+import omero.cmd.IRequest;
+import omero.cmd.Response;
+import omero.model.Session;
+import omero.util.IceMapper;
+import edu.emory.mathcs.backport.java.util.Collections;
+
+@SuppressWarnings("serial")
+public class CurrentSessionsRequestI extends CurrentSessionsRequest
+    implements IRequest {
+
+    protected Helper helper;
+
+    protected final CurrentDetails current;
+
+    protected final SessionManager manager;
+
+    protected Map<String, EventContext> contexts;
+
+    public CurrentSessionsRequestI(CurrentDetails current,
+            SessionManager manager) {
+        this.current = current;
+        this.manager = manager;
+    }
+
+    //
+    // CMD API
+    //
+
+    @Override
+    public Map<String, String> getCallContext() {
+        return null;
+    }
+
+    public void init(Helper helper) {
+        this.helper = helper;
+        this.helper.setSteps(1);
+    }
+
+    public Object step(int step) throws Cancel {
+        helper.assertStep(step);
+
+        contexts = manager.getAll();
+        if (contexts.size() == 0) {
+            return Collections.emptyList();
+        }
+        return helper.getServiceFactory().getQueryService().
+                findAllByQuery("select s from Session s where s.uuid in (:uuid)",
+                        new Parameters().addList("uuid", new ArrayList<String>(
+                                contexts.keySet())));
+    }
+
+    @Override
+    public void finish() throws Cancel {
+        // no-op
+    }
+
+    public void buildResponse(int step, Object object) {
+        helper.assertResponse(step);
+
+        @SuppressWarnings("unchecked")
+        List<ome.model.meta.Session> rv = (List<ome.model.meta.Session>) object;
+        Map<String, Session> objects = new HashMap<String, Session>();
+        IceMapper mapper = new IceMapper();
+        for (ome.model.meta.Session obj : rv) {
+            objects.put(obj.getUuid(), (Session) mapper.map(obj));
+        }
+
+        if (helper.isLast(step)) {
+            CurrentSessionsResponse rsp = new CurrentSessionsResponse();
+            rsp.sessions = new ArrayList<Session>(contexts.size());
+            rsp.contexts = new ArrayList<omero.sys.EventContext>(contexts.size());
+            for (Map.Entry<String, EventContext> entry : contexts.entrySet()) {
+                String uuid = entry.getKey();
+                Session s = objects.get(uuid);
+                rsp.sessions.add(s);
+                if (s == null) {
+                    // Non-admin
+                    EventContext orig = entry.getValue();
+                    omero.sys.EventContext ec = new omero.sys.EventContext();
+                    rsp.contexts.add(ec);
+                    ec.userId = orig.getCurrentUserId();
+                    ec.userName = orig.getCurrentUserName();
+                    ec.groupId = orig.getCurrentGroupId();
+                    ec.groupName = orig.getCurrentGroupName();
+                    ec.isAdmin = orig.isCurrentUserAdmin();
```

Comment:
No, it's not a warnable offense. This basically just returns the "safe" state for other users, i.e. what's inferable from an HQL call.